sonar-dotnet-shared-library/src/main/java/org/sonarsource/dotnet/shared/plugins/SarifParserCallbackImpl.java

```diff
@@ -0,0 +1,124 @@
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonarsource.dotnet.shared.plugins;
+
+import java.util.Collection;
+import java.util.Map;
+import org.sonar.api.batch.fs.InputFile;
+import org.sonar.api.batch.sensor.SensorContext;
+import org.sonar.api.batch.sensor.issue.NewIssue;
+import org.sonar.api.batch.sensor.issue.NewIssueLocation;
+import org.sonar.api.rule.RuleKey;
+import org.sonarsource.dotnet.shared.sarif.Location;
+import org.sonarsource.dotnet.shared.sarif.SarifParserCallback;
+
+public class SarifParserCallbackImpl implements SarifParserCallback {
+  private final SensorContext context;
+  private final Map<String, String> repositoryKeyByRoslynRuleKey;
+
+  public SarifParserCallbackImpl(SensorContext context, Map<String, String> repositoryKeyByRoslynRuleKey) {
+    this.context = context;
+    this.repositoryKeyByRoslynRuleKey = repositoryKeyByRoslynRuleKey;
+  }
+
+  @Override
+  public void onProjectIssue(String ruleId, String message) {
+    String repositoryKey = repositoryKeyByRoslynRuleKey.get(ruleId);
+    if (repositoryKey == null) {
+      return;
+    }
+    NewIssue newIssue = context.newIssue();
+    newIssue
+      .forRule(RuleKey.of(repositoryKey, ruleId))
+      .at(newIssue.newLocation()
+        .on(context.module())
+        .message(message))
+      .save();
+  }
+
+  @Override
+  public void onFileIssue(String ruleId, String absolutePath, String message) {
+    String repositoryKey = repositoryKeyByRoslynRuleKey.get(ruleId);
+    if (repositoryKey == null) {
+      return;
+    }
+
+    InputFile inputFile = context.fileSystem().inputFile(context.fileSystem().predicates()
+      .hasAbsolutePath(absolutePath));
+    if (inputFile == null) {
+      return;
+    }
+
+    NewIssue newIssue = context.newIssue();
+    newIssue
+      .forRule(RuleKey.of(repositoryKey, ruleId))
+      .at(newIssue.newLocation()
+        .on(inputFile)
+        .message(message))
+      .save();
+  }
+
+  @Override
+  public void onIssue(String ruleId, Location primaryLocation, Collection<Location> secondaryLocations) {
+    String repositoryKey = repositoryKeyByRoslynRuleKey.get(ruleId);
+    if (repositoryKey == null) {
+      return;
+    }
+
+    InputFile inputFile = context.fileSystem().inputFile(context.fileSystem().predicates()
+      .hasAbsolutePath(primaryLocation.getAbsolutePath()));
+    if (inputFile == null) {
+      return;
+    }
+
+    NewIssue newIssue = context.newIssue();
+    newIssue
+      .forRule(RuleKey.of(repositoryKey, ruleId))
+      .at(newIssue.newLocation()
+        .on(inputFile)
+        .at(inputFile.newRange(primaryLocation.getStartLine(), primaryLocation.getStartColumn(),
+          primaryLocation.getEndLine(), primaryLocation.getEndColumn()))
+        .message(primaryLocation.getMessage()));
+
+    for (Location secondaryLocation : secondaryLocations) {
+      if (!inputFile.absolutePath().equalsIgnoreCase(secondaryLocation.getAbsolutePath())) {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this use of "absolutePath"; it is deprecated. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)